### PR TITLE
Fix warnings in codegen

### PR DESCRIPTION
**Pull request by FrankBakkerNl**
## Proposed change

Fixed Nullable warnings in codegen by making properties nullable

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code compiles without warnings (code quality chek)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

File: src/HassModel/NetDaemon.HassModel.CodeGenerator/Generator.Entitites.cs [hunk 1]
```diff
@@ -103,7 +103,7 @@ private static IEnumerable<TypeDeclarationSyntax> GenerateEntityAttributeRecords
                     d => { d.Name += i++; return d; });
 
                 var autoProperties = autoPropertiesParams.Select(a =>
-                    Property(a.TypeName, a.Name).ToPublic().WithAttribute<JsonPropertyNameAttribute>(a.SerializationName))
+                    Property(a.TypeName + "?", a.Name).ToPublic().WithAttribute<JsonPropertyNameAttribute>(a.SerializationName))
```
Comment (BeeHiveJava) [inline src/HassModel/NetDaemon.HassModel.CodeGenerator/Generator.Entitites.cs:106]: Have you considered using string interpolation for these? I don't have a preference, but we use interpolation in other places here as well.

File: src/HassModel/NetDaemon.HassModel.Tests/CodeGenerator/CodeGeneratorTest.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@ public void RunCodeGenEMpy()
 
             code.DescendantNodes().OfType<NamespaceDeclarationSyntax>().First().Name.ToString().Should().Be("RootNameSpace");
             
-            AssertCodeCompiles(code.ToString());
+            AssertCodeCompiles(code.ToString(), "");
```
Comment (BeeHiveJava) [inline src/HassModel/NetDaemon.HassModel.Tests/CodeGenerator/CodeGeneratorTest.cs:25]:
```suggestion
            AssertCodeCompiles(code.ToString(), string.Empty);
```

File: src/HassModel/NetDaemon.HassModel.CodeGenerator/Generator.Entitites.cs (hunk 1 above)
Comment (helto4real) [reply src/HassModel/NetDaemon.HassModel.CodeGenerator/Generator.Entitites.cs:106]: I just think `$"{a.TypeName}?"` looks better. Using + between strings used to be less efficient than other methods but that probably does not comply. For this case I am fine with having it as is too.

**Review by helto4real — APPROVED** [review]
Thanks!

**Comment by helto4real** [comment]
I approved it. If you wanna make the changes as commented I left it unmerged but it is nothing big.

**Pull request merged**